src/main/java/controller/MainController.java

```diff
@@ -1,241 +1,331 @@
+	 * 
+	 * @return JTable
+	 */
+	private JTable createExcelTable() {
+		String[][] dataForTable = new String[excelRows.size() - 1][excelRows.get(1).length];
+
+		for (int i = 0; i < dataForTable.length; i++) {
+			for (int j = 0; j < dataForTable[i].length; j++) {
+				dataForTable[i][j] = excelRows.get(i + 1)[j].toString();
+			}
+		}
+
+		return new JTable(dataForTable, excelRows.get(0));
+	}
+
+	/**
+	 * Converts all the valid rows into ExcelRow model
+	 * 
+	 * Starts at index 1 because we only want to convert the table content and not
+	 * the header
+	 */
+	private void convertExcelRows() {
+		for (int i = 1; i < excelRows.size(); i++) {
+			try {
+				excelRowsConverted.add(new ExcelRow(excelRows.get(i)));
+			}
+
+			catch (Exception e) {
+				JOptionPane.showMessageDialog(null, "An error occurred while converting the Excel Rows", "Warning",
+						JOptionPane.WARNING_MESSAGE);
+				throw e;
+			}
+		}
+	}
+
+	/**
+	 * This method is used to open the EditRuleController which controls the Rule
+	 * Edition GUI
+	 */
+	private void editButton(JButton editButton, JComboBox<CodeQualityRule> ruleListBox) {
+
+		editButton.addActionListener(e -> {
+
+			CodeQualityRule rule = (CodeQualityRule) ruleListBox.getSelectedItem();
+
+			if (rule == null) {
+				new EditRuleController();
+
+			} else {
+				new EditRuleController(rule);
+			}
+		});
+	}
+
+	/**
+	 * Sets the add Button in the MainFrame to open an empty Rule Edition Popup
+	 */
+	private void addButton(JButton addButton) {
+		addButton.addActionListener(e -> new EditRuleController());
+	}
+
+	/**
+	 * Verify the code quality based on the Rules created and sends the results to
+	 * be displayed in the QualityRulesResultFrame
+	 * 
+	 * @throws ScriptException
+	 */
+	private void checkCodeQualityAndShow() {
+		String[][] results = null;
+		results = getCodeQualityResults();
+		String[] colNames = new String[5 + rulesList.size()];
+		colNames[0] = "Method ID";
+		colNames[1] = "long_method";
+		colNames[2] = "feature_envy";
+		colNames[3] = "PMD";
+		colNames[4] = "iPlasma";
+		int iterator = 5;
+		for (CodeQualityRule rule : rulesList) {
+			colNames[iterator] = rule.getName();
+			iterator++;
+		}
+
+		if (results != null) {
+			qualityGui.fillTable(results, colNames);
+			qualityGui.show();
+		}
+	}
+
+	/**
+	 * 
+	 * Returns the results of the calculation of each rule, for each method.
+	 * 
+	 * @return An Array of String arrays where each line is a row with the code
+	 *         quality results for a method, and each column is the value of that
+	 *         result line for that column
+	 */
+	private String[][] getCodeQualityResults() {
+		String[][] results = new String[excelRowsConverted.size()][5 + rulesList.size()];
+		int iterator = 0;
+		for (ExcelRow row : excelRowsConverted) {
+			String[] qualityRow = new String[4 + excelRowsConverted.size()];
+			qualityRow[0] = Integer.toString(row.getId());
+			qualityRow[1] = Boolean.toString(row.isIs_long_method());

```

Comment:
this is missing the refactor right? the one diego did?

Reply:
It's possible, this branch was created a couple of days ago. Which variables did he refactor?

Reply:
basically everything - it's already in the master branch, you can merge into yours and solve the conflicts